tests/src/Context/DrupalContext.php

```diff
@@ -209,4 +216,111 @@ public function theFollowingContentsUsingForWysiwygFields($text_format, TableNod
     return $nodes;
   }
 
+  /**
+   * Add a field to a specified content type.
+   *
+   * @param string $arg1
+   *    The machine name of the content type to which attaching the field.
+   * @param TableNode $settings
+   *    Two columns table containing field settings.
+   *
+   * @Given a field with the following settings is added to :arg1 type:
+   *
+   * @throws \InvalidArgumentException
+   *    If the field exist already in the Drupal system.
+   */
+  public function aFieldWithTheFollowingSettingsIsAddedToType($arg1, TableNode $settings) {
+    $field_info = array();
+    $field_instance_info = array();
+
+    $field_instance_info['entity_type'] = 'node';
+    $field_instance_info['bundle'] = $arg1;
+
+    // Assign fields to user before creation.
+    foreach ($settings->getRowsHash() as $setting_name => $value) {
+      switch ($setting_name) {
+        case 'Label':
+          $field_instance_info['label'] = $value;
+          break;
+
+        case 'Name':
+          $field_name = 'field_' . $value;
+          $this->createFieldList[$field_name] = array(
+            'field_name' => $field_name,
+            'entity_type' => 'node',
+            'bundle' => $arg1,
+          );
+          $is_field_exiting = field_info_field($field_name);
+          if ($is_field_exiting) {
```

Comment:
The purpose of this code is purely technical. it is to avoid that someone creates a field definition that could change the configuration of an field of the platform. According to me, that will be useful while defining new scenarii that use this step.